[cmake][addons] change binary addon includes cmake install way

AlwinEsch opened the pull request:
## Description

This change the way how the headers of binary addons becomes
installed from Kodi. Instead to define every file becomes the
folder with headers installed. All includes who are outside
from the dev-kit place are still installed by the old way.

Reasons of this changes are:
- Allow to copy header who defined in different include folders,
e.g. the coming:
```cpp
#include <kodi/addon-instance/PVRClient.h>
#include <kodi/gui/DialogOK.h>
```
- Make it more easy during add of new header in dev-kit, before was always
a update on cmake needed.

The only thing to be noticed is that the dev-kit include only headers.
This was also the main reason that this folder was made and did not give
it as hard.

<!--- Provide a general summary of your change in the Title above -->

<!--- Describe your change in detail -->

## Motivation and Context
Next part from kodi-agile
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch commented:
Is that OK and have not forgotten anything?

hudokkow commented on cmake/cpack/deb/packages/kodi-addon-dev.txt.in:
```diff
@@ -16,7 +16,7 @@ PACKAGE_SECTION libdevel
+PACKAGE_RECOMMENDS
 PACKAGE_SUGGESTS
 PACKAGE_BREAKS
 PACKAGE_REPLACES
```
I think `PACKAGE_REPLACES` should be populated with `@APP_NAME_LC@-audio-dev, @APP_NAME_LC@-inputstream-dev, @APP_NAME_LC@-pvr-dev, @APP_NAME_LC@-screensaver-dev, @APP_NAME_LC@-visualization-dev`

@wsnipex to confirm ^^

wsnipex replied:
correct. If you replace all the different packages with only kodi-addon-dev, this needs to replace the old packages

hudokkow commented on xbmc/addons/addon-bindings.mk:
```diff
@@ -1,45 +1,11 @@
+# List here contains all addon related header who are not defined in
+# ./addons/kodi-addon-dev-kit/include/kodi
```
List contains only add-on related headers not inside

hudokkow reviewed (commented):
Looks good to me, apart from the minors pointed.
Give me a few minutes and I'll test the PR.

wsnipex commented:
this will require changes for PPA packaging.

hudokkow reviewed (approved):
Tested and working 100%. Just needs to minors corrected.
:+1:

AlwinEsch committed:
[cmake][addons] change binary addon includes cmake install way
This change the way how the headers of binary addons becomes
installed from Kodi. Instead to define every file becomes the
folder with headers installed. All includes who are outside
from the dev-kit place are still installed by the old way.

Reasons of this changes are:
- Allow to copy header who defined in different include folders,
e.g. the coming:
```cpp
\#include <kodi/addon-instance/PVRClient.h>
\#include <kodi/gui/DialogOK.h>
```
- Make it more easy during add of new header in dev-kit, before was always
a update on cmake needed.

The only thing to be noticed is that the dev-kit include only headers.
This was also the main reason that this folder was made and did not give
it as hard.

AlwinEsch committed:
[cmake] update debian package files
With only one dev kit for everything are several parts no more needed.

AlwinEsch commented:
is updated

jenkins build this please

AlwinEsch commented:
jenkins build this please

wsnipex commented:
who is going to fix debian packaging in every addon?

hudokkow commented:
Me? 😆

AlwinEsch commented:
I can also do it :-D Am anyway in addon merge mood

notspiff commented:
globbing for headers. hrmf.

AlwinEsch commented:
Hi @notspiff what exactly you mean?
1. Bad to use folder instead of addon-bindings.mk
2. Better to use `file(GLOB_RECURSE` instead `file(copy`

notspiff commented:
1. It's a generic principle I follow, I never glob for sources, it makes things unpredictable. No biggie, just a knee jerk. It's probably more convenient than dangerous in the context.

AlwinEsch commented:
I give you partly right. But to handle includes in different destination folders has produced some problems. Also problematic becomes then that in `cmake/scripts/linux/Install.cmake` it is defined as files.

This Install.cmake is for me really a bad way and need to be improved in future. That's why I also like to have only one debian dev package there who can use addon-bindings.mk.

I can try to create a macro to read a modified addon-bindings.mk and handle them also in Install.cmake.
